easybuild/framework/easyblock.py

```diff
@@ -965,6 +965,15 @@ def make_module_dep(self, unload_info=None):
         # if the modules that extend $MODULEPATH are not loaded this module is not available, so there is not
         # point in loading them again (in fact, it may cause problems when reloading this module due to a load storm)
         deps = [d for d in deps if d not in excluded_deps]
+
+        # load modules that open up the module tree before checking deps of deps (in reverse order)
+        self.modules_tool.load(excluded_deps[::-1])
+ 
+        for dep in excluded_deps:
```

Comment:
@bartoldeman one tine suggestion: I would rename `dep` to `excluded_dep` here, to avoid confusion...